phoenix-core/src/main/java/org/apache/phoenix/parse/ShowCreateTablesStatement.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.phoenix.parse;
+
+import org.apache.phoenix.compile.ColumnResolver;
+import org.apache.phoenix.thirdparty.com.google.common.base.Preconditions;
+import org.apache.phoenix.util.SchemaUtil;
+
+import java.util.Objects;
+
+/**
+ * ParseNode implementation for DESCRIBE tablename.
+ */
+public class ShowCreateTablesStatement extends ShowCreateStatement {
+
+    private TableName tableName;
+
+    public ShowCreateTablesStatement(TableName tn) {
+        tableName = tn;
+    }
+
+    public TableName getTableName() {
+        return tableName;
+    }
+
+    public void toSQL(ColumnResolver resolver, StringBuilder buf) {
+        Preconditions.checkNotNull(buf);
+        buf.append("SHOW CREATE STATEMENT ");
+
+        if(tableName.getSchemaName() != null) {
+            if (tableName.isSchemaNameCaseSensitive()) {
+                buf.append("\"");
+            }
+            buf.append(tableName.getSchemaName());
+            if (tableName.isSchemaNameCaseSensitive()) {
+                buf.append("\"");
+            }
+            buf.append(".");
+        }
+
+        if (tableName.isTableNameCaseSensitive()) {
+            buf.append("\"");
+        }
+        buf.append(tableName.getTableName());
+        if (tableName.isTableNameCaseSensitive()) {
+            buf.append("\"");
+        }
```

Comment:
Pretty sure we have a helper function already for this.

Reply:
I updated this to use the SchemaUtil.getEscapedTableName, its not exactly the same but works.